utility/getPipelineStages.py

```diff
@@ -0,0 +1,135 @@
+
+    Usage:
+        utility/getPipelineStages.py --rhcephVersion <VER> --tags <tags>
+                  [--overrides <str>]
+                  [--metadata <FILE>]
+
+        utility/getPipelineStages.py (-h | --help)
+
+    Options:
+        -h --help          Shows the command usage
+        -v --rhcephVersion VER     The rhcephVersion for which test stages need to be fetched
+        -t --tags <str>    tags to be used for filtering test scripts
+        -o --overrides <str>       Overrides to be considered for execution
+        -m --metadata FILE         Metadata file full path if it is not present in default pipeline/metadata folder
+"""
+
+
+def load_file(file_name):
+    """Retrieve yaml data content from file."""
+    file_path = os.path.abspath(file_name)
+    with open(file_path, "r") as conf_:
+        content = yaml.safe_load(conf_)
+    return content
+
+
+def generate_random_string(length):
+    """Generate a random alphanumeric string of given length"""
+    return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+
+def fetch_stages(args):
+    """
+    Fetch test stages to be executed based on the input provided
+    Args:
+        args:
+            Dictionary containing the tier_level, cloud_type and all other necessary override parameters
+            to be used to fetch test stages for execution
+
+            Examples:
+                arguments = {
+                    "rhcephVersion": <the rhbuild version for which the stages need to be fetched>,
+                    "tags": <list of strings containing the tags based on which the stages should be filtered
+                            ex: [nightly, tier-1, psi, stage-1, dmfg]>,
+                    "overrides": <overrides in dict format, all keys accepted by run.py are supported to be overridden
+                                ex: {"cloud_type": "psi", "tier": "tier-1", "platform": "rhel-8"}>,
+                                if any key similar to --store which does not have a value it can be passed as
+                                "store": ""
+                    "metadata_file": <path to the metadata file>,
+                }
+
+    Returns:
+        Yaml string containing CLI to be executed for each test suite at every stage
+        Examples:
+            "stage-1":
+                "test-cephfs-core-features":
+                    "execute_cli": "<workspace>/.venv/bin/python run.py ......",
+                     "cleanup_cli": "...."
+                .....
+            ......
+    """
+    rhceph_version = args.get("rhcephVersion")
+    metadata_file = args.get("metadata_file")
+    if not rhceph_version and not metadata_file:
+        log.error(
+            "Cannot fetch test stages since rhcephVersion and metadata file are not specified"
+        )
+    if not metadata_file:
+        metadata_file = f"pipeline/metadata/{rhceph_version}.yaml"
+    data = load_file(metadata_file)
+    if args.get("tags"):
```

Comment:
Exception will be raised at line #88 in case `tags` is not passed as variable won't get declared. Below code can avoid same -
```
tags = args.get("tags", "").split(",")
```

this will return `''` even in case `tags` not passed, but variable get declared all the time

Reply:
Done.